src/components/ha-selector/ha-selector.ts

```diff
@@ -0,0 +1,64 @@
+import { customElement, html, LitElement, property } from "lit-element";
+import { dynamicElement } from "../../common/dom/dynamic-element-directive";
+import { HomeAssistant } from "../../types";
+
+import "./ha-selector-entity";
+import "./ha-selector-device";
+
+export type Selector = EntitySelector | DeviceSelector;
+
+export interface EntitySelector {
```

Comment:
I think that we should move these to `data/selector.ts` since they will be referenced by other types.